scripts/provision.sh

```diff
@@ -0,0 +1,260 @@
+	$terraform_output_bosh_network_name \
+	$terraform_output_bosh_ip
+    fi
+  fi
+  if ! bosh_check_and_login; then
+    echo "Failed to contact BOSH node $BOSH_IP:$BOSH_PORT after provisioning"
+  fi
+
+}
+
+clone_and_update_cf_release(){
+  # Git clone and upload release
+  echo "Updating ~/cf-release from $CF_RELEASE_GIT_URL:$CF_RELEASE_REVISION"
+
+  if [ ! -d ~/cf-release/.git ]; then
+    rm -rf ~/cf-release
+    git clone -q $CF_RELEASE_GIT_URL ~/cf-release
+  else
+    cd ~/cf-release
+    git remote set-url origin $CF_RELEASE_GIT_URL
+    git fetch -q
+  fi
+
+  cd ~/cf-release
+  git checkout -q $CF_RELEASE_REVISION
+
+  ./update  >> update.log 2>&1
+  if [ $? != 0 ]; then
+    echo "Update failed, check ~/cf-release/update.log for details: "
+    tail update.log
+  fi
+}
+
+upload_stemcell() {
+  # Download the stemcell if it is not locally
+  cd ~
+  if [ ! -f $STEMCELL ]; then
+    echo "Downloading stemcell $STEMCELL"
+    if [ "$STEMCELL_URL" ]; then
+      wget $STEMCELL_URL
+    else
+      time bosh download public stemcell $STEMCELL
+    fi
+  fi
+
+  # Extract stemcell version and name info
+  mkdir -p /tmp/{$STEMCELL}.d
+  tar -xzf $STEMCELL -C /tmp/{$STEMCELL}.d stemcell.MF
+  stemcell_name=$(cat /tmp/{$STEMCELL}.d/stemcell.MF | awk '/^name:/ { print $2 }')
+  stemcell_version=$(cat /tmp/{$STEMCELL}.d/stemcell.MF | awk '/^version:/ { print $2 }' | tr -d "'")
+
+  # Upload stemcell if it is not uploaded
+  if $SCRIPT_DIR/bosh_list_stemcells.rb | grep -q -e "$stemcell_name/$stemcell_version"; then
+    echo "Stemcell $stemcell_name/$stemcell_version already uploaded, skipping"
+  else
+    time bosh upload stemcell $STEMCELL --skip-if-exists
+  fi
+}
+
+upload_releases() {
+  for r in $BOSH_RELEASES; do
+    local name=$(echo $r | cut -f 1 -d ,)
+    local version=$(echo $r | cut -f 2 -d ,)
+    local url=$(echo $r | cut -f 3 -d ,)
+    if $SCRIPT_DIR/bosh_list_releases.rb | grep -q "$name/$version"; then
+      echo "Release $name version $version already uploaded, skipping"
+      continue
+    else
+      bosh upload release $url 2>&1 | tee /tmp/upload_release.log
+      if [ $PIPESTATUS != 0 ] && ! grep -q -e 'Release.*already exists' /tmp/upload_release.log;  then
+        return 1
+      fi
+    fi
+  done
+}
+
+cf_prepare_deployment() {
+  clone_and_update_cf_release
+  upload_stemcell
+  upload_releases
+}
+
+cf_compile_manifest() {
+  # Use spiff to generate CF deployment manifest
+  cd ~
+
+  # Output the director uuid to be populated by spiff
+  echo -e "---\ndirector_uuid: $(bosh status --uuid)" > templates/stubs/director-uuid.yml
+
+  # Generate the manifest
+  CF_RELEASE_PATH=~/cf-release/ ./generate_deployment_manifest.sh $TARGET_PLATFORM > ~/cf-manifest.yml
+}
+
+cf_deploy() {
+  bosh deployment ~/cf-manifest.yml
+  time bosh -n deploy
+}
+
+cf_post_deploy() {
+  # Deploy psql broker
+  time bash $SCRIPT_DIR/deploy_psql_broker.sh admin fakepassword
+}
+
+install_dependencies
```

Comment:
Create a chain with `cmd1 && cmd2 && cmd3...` so the script stops processing if an error is encountered.